components/up.libsonnet

```diff
@@ -109,6 +86,93 @@ local k = import 'ksonnet/ksonnet.beta.4/k.libsonnet';
     },
   },
 
+  withReadEndpoint:: {
+    local u = self,
+    readEndpoint:: error 'must provide read endpoint',
```

Comment:
Following our existing config pattern, I believe this should be in the config.

Reply:
Yeah I see that, but is it not better to set these only when used? otherwise you can easily forget to remove it when you don't need them. With this approach it is explicit and goes hand in hand with the actual mixin.